src/backend/distributed/utils/colocation_utils.c

```diff
@@ -660,6 +663,19 @@ UpdateRelationColocationGroup(Oid distributedRelationId, uint32 colocationId)
 
 	systable_endscan(scanDescriptor);
 	heap_close(pgDistPartition, NoLock);
+
+	shouldSyncMetadata = ShouldSyncTableMetadata(distributedRelationId);
+	if (shouldSyncMetadata)
+	{
+		StringInfo command = makeStringInfo();
+		char *relationName = get_rel_name(distributedRelationId);
+		appendStringInfo(command, "UPDATE pg_dist_partition "
+								  "SET colocationid='%d' "
+								  "WHERE logicalrelid='%s'::regclass",
```

Comment:
We should never be doing the escaping in the format string, see metadata_sync.c for examples.